frontend/app/(defaultLayout)/app-router/Store/[[...slug]]/data.ts

```diff
@@ -0,0 +1,8 @@
+import 'server-only';
```

Comment:
Do we need to install this package in order to use it? It appears that the app runs fine without it 🤷🏻 

https://nextjs.org/docs/app/building-your-application/rendering/composition-patterns#keeping-server-only-code-out-of-the-client-environment

Reply:
I'll double check it because I might have read a different part of the docs 🤔

Reply:
@masonmcelvain installed the package for good measure. I thought it was just sort of a Next.js pragma like `"use client"`, but apparently it's a package released by react team https://www.npmjs.com/package/server-only

Reply:
🤔 I don't understand why several other packages changed in `pnpm-lock.yaml` when we install `server-only`, and I get the same result locally. CI failed because it could not build the app

```
- info Collecting page data...
TypeError: Class extends value undefined is not a constructor or null
    at 86404 (/home/runner/work/react-commerce/react-commerce/frontend/.next/server/chunks/43.js:77701:45)
    at __webpack_require__ (/home/runner/work/react-commerce/react-commerce/frontend/.next/server/webpack-runtime.js:25:43)
    at 33839 (/home/runner/work/react-commerce/react-commerce/frontend/.next/server/chunks/43.js:77641:18)
    at __webpack_require__ (/home/runner/work/react-commerce/react-commerce/frontend/.next/server/webpack-runtime.js:25:43)
    at 60979 (/home/runner/work/react-commerce/react-commerce/frontend/.next/server/chunks/43.js:106463:15)
    at __webpack_require__ (/home/runner/work/react-commerce/react-commerce/frontend/.next/server/webpack-runtime.js:25:43)
    at 98100 (/home/runner/work/react-commerce/react-commerce/frontend/.next/server/chunks/554.js:12177:14)
    at __webpack_require__ (/home/runner/work/react-commerce/react-commerce/frontend/.next/server/webpack-runtime.js:25:43)
    at 37265 (/home/runner/work/react-commerce/react-commerce/frontend/.next/server/chunks/554.js:9321:72)
    at __webpack_require__ (/home/runner/work/react-commerce/react-commerce/frontend/.next/server/webpack-runtime.js:25:43)

> Build error occurred
Error: Failed to collect page data for /products/[handle]
    at /home/runner/work/react-commerce/react-commerce/node_modules/.pnpm/next@13.4.12_@babel+core@7.17.9_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/build/utils.js:1[156](https://github.com/iFixit/react-commerce/actions/runs/7745969669/job/21123382036#step:8:159):15
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  type: 'Error'
}
```

If it works without installing this package, then perhaps it's easiest to omit it for now.

Reply:
Yeah the build of locally and on Vercel works, but for some reason not in CI. Will look into this a bit more

Reply:
@masonmcelvain for now `server-only` is just a nice to have, so I've removed it. Will look into this later